crates/stackable-operator/src/commons/affinity.rs

```diff
@@ -51,6 +60,11 @@ pub struct StackableNodeSelector {
 
 impl Atomic for StackableNodeSelector {}
 
+/// We need a custom JsonSchema for [`StackableNodeSelector`], please have a look at the documentation there.
+pub fn optional_stackable_node_selector_schema(gen: &mut schemars::gen::SchemaGenerator) -> Schema {
+    Option::<BTreeMap<String, String>>::json_schema(gen)
+}
```

Comment:
This could be an explicit `impl JsonSchema for StackableNodeSelector`, which should make this non-breaking

Reply:
Good to know!